core/mount/mount_linux.go

```diff
@@ -244,11 +249,9 @@ func getUnprivilegedMountFlags(path string) (int, error) {
 	return flags, nil
 }
 
-func doPrepareIDMappedOverlay(lowerDirs []string, usernsFd int) (tmpLowerDirs []string, _ func(), _ error) {
-	td, err := os.MkdirTemp(tempMountLocation, "ovl-idmapped")
```

Comment:
what about keeping this line in this function? I was thinking that no need to let caller prepare this folder.

Reply:
This just simplified testing `doPrepareIDMappedOverlay`, this way unit tests can create directories that are cleaned up by the test runtime. It's a private function with only one non-test caller. But maybe the approach to tests will be redone.